src/treelearner/cuda_tree_learner.cpp

```diff
@@ -63,6 +63,43 @@ CUDATreeLearner::CUDATreeLearner(const Config* config)
 }
 
 CUDATreeLearner::~CUDATreeLearner() {
+  #pragma omp parallel for schedule(static, num_gpu_)
```

Comment:
With a static schedule, and `num_gpu_` chunk size, I think there will be only 1 thread being used. So I can we simply remove this line?

Reply:
@shiyu1994 
To be honest I've just copy-pasted simplified code from [here](https://github.com/microsoft/LightGBM/blob/master/src/treelearner/cuda_tree_learner.cpp#L319) which uses `omp parallel`, so I decided that probably it is done intentionally. Do you think we should actually remove this?

Reply:
After experimenting, I find that schedule(static, num_gpu_) will only result in a single thread. With `num_gpu_` block size, and with `num_gpu_` total iterations. So this would not parallelize the for loop.
I think it is ok to keep the `omp parallel` in this PR. And we can fix this with another PR. Thank you!